protocol/syncer.go

```diff
@@ -379,21 +387,26 @@ func (s *Syncer) findCommonAncestor(clt proto.V1Client, status *Status) (*types.
 				min = m + 1
 			} else {
 				if m == 0 {
-					return nil, nil, fmt.Errorf("genesis does not match?")
+					return nil, nil, errors.New("genesis does not match?")
 				}
 				max = m - 1
 			}
 		}
 	}
 
+	var num uint64
+	if header != nil {
```

Comment:
`header` can be nil. How to handle this?

Reply:
As it seems like the same type of problem as with being unable to find a certain header by number or with being unable to read the local genesis, I'd advise: 

```
		return nil, nil, errors.New("header is nil")
```

Also, can you please clarify why did you keep `errors.New()` in this case, while keeping the`fmt.Errorf()` above?  Thanks
@nicovogelaar